app/models/event.rb

```diff
@@ -1,7 +1,57 @@
 # frozen_string_literal: true
 
+BOOLEAN = [true, false].freeze
+
 class Event < ApplicationRecord
-  validates :outbound_url, presence: true
+  before_validation :set_urls
+  validates :test, inclusion: { in: BOOLEAN, message: '"test" field must be true or false' }
+  validates :completed, inclusion: [true, false]
+  validates :status_code, numericality: { greater_than_or_equal_to: 100, less_than_or_equal_to: 599 }, allow_nil: true
+  validate :completed_at_format
+  validate :data_json_object
 
   belongs_to :bridge
+
+  def sidebar_format
+    updated_at = String(self.updated_at)
+    time = date_format(updated_at.split(' ')[1])
+    date = updated_at.split(' ')[0]
+    { id: id,
+      time: time.slice(0..-3),
```

Comment:
Why date & time? Why not dateTime? I can format it on the frontend. JS is much faster than ruby too!

Reply:
That was actually for frontend user friendliness, so that the four pieces of info you need each have a key. But no problem if you want to handle in on that end.